cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -618,12 +630,40 @@ public Secret generateBrokersSecret() {
         return portList;
     }
 
-    private List<Volume> getVolumes(boolean isOpenShift) {
-        List<Volume> volumeList = new ArrayList<>();
+    /**
+     * Fill the with volumes, persistent volume claims and related volume mount paths for the storage
+     * It's called recursively on the related inner volumes if the storage is of {@link Storage#TYPE_JBOD} type
+     *
+     * @param storage the Storage instance from which building volumes, persistent volume claims and related volume mount paths
+     */
+    private void setDataVolumesClaimsAndMountPaths(Storage storage) {
+        Integer id = null;
         if (storage instanceof EphemeralStorage) {
-            volumeList.add(createEmptyDirVolume(VOLUME_NAME));
+            id = ((EphemeralStorage) storage).getId();
+        } else if (storage instanceof PersistentClaimStorage) {
+            id = ((PersistentClaimStorage) storage).getId();
+        } else if (storage instanceof JbodStorage) {
+            for (SingleVolumeStorage volume: ((JbodStorage) storage).getVolumes()) {
+                setDataVolumesClaimsAndMountPaths(volume);
+            }
+            return;
         }
 
+        String name = id != null ? VOLUME_NAME + "-" + id : VOLUME_NAME;
```

Comment:
I would put this in a method.

Reply:
It's not reused in any other place, why do you think it deserves a method?

Reply:
1) You get to javadoc what the naming convention will be. I.e. there's more of a contract.
2) Future refactorings will tend to just call the method. Without the method future refactorings will tend to copy paste. Let me illustrate by pointing to all the places in the code base where we use things like `ss.getMetadata().getName() + "-" + podNum`. 
3) Because the path will be visible in the reassigning partitions API we should probably put this method in KafkaResources, or at least somewhere in the api module.
4) If everywhere is using the method I get to use "find callers" in my IDE when reasoning about the code in the future. Without it I'm left having to use craft a `grep` which wastes of time.

Reply:
Thinking more I have to change this one, the name should be always something like `data-<id>` even when user has just `type=persistent-claim` (so it will be `data-1`) because if the user wants to move from P to J [P] the `name` has to be the same so we cannot move from `data` to `data-1` but we should stick on `data-1`.